### ISSUE TITLE
Error Occurs When Negative Parameter is Passed to epoch_to_timestamp Function.

### ISSUE THREAD

**Issue by yjhong79** [pacman82/arrow-odbc#111]
When passing a negative value to the epoch_to_timestamp function, the following error occurs:

```
called `Result::unwrap()` on an `Err` value: TryFromIntError(())
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```
Values before January 1, 1970, are negative. This means that any timestamp representing a date prior to the Unix epoch (1970-01-01) will be a negative number. The function should ideally handle these cases without resulting in an error.

For example, test with the following code:
```
_ = epoch_to_timestamp::<1_000_000>(-1);
```

Thank you.

**Comment by pacman82** [comment]
I assume you have happend to discover that behavior then trying to read data with timestamps older than 1970-01-01? What is the ask here? I think yielding a proper error instead of a panic is doable, yet supporting such timestamps in arrow is a different beast.

Do you want to get an error instead of the panic? Or is it something else you are after?

**Comment by yjhong79** [comment]
I wrote a test program and obtained the following results.
``` rust
use arrow::{
    array::{
        RecordBatch, TimestampMicrosecondArray, TimestampMillisecondArray,
        TimestampNanosecondArray, TimestampSecondArray,
    },
    datatypes::{DataType, Field, Schema, TimeUnit},
};

fn main() {
    let sec_array = TimestampSecondArray::from(vec![-1]);
    let mili_array = TimestampMillisecondArray::from(vec![-1]);
    let micro_array = TimestampMicrosecondArray::from(vec![-1]);
    let nano_array = TimestampNanosecondArray::from(vec![-1]);
    let schema = Schema::new(vec![
        Field::new("sec", DataType::Timestamp(TimeUnit::Second, None), false),
        Field::new(
            "mili",
            DataType::Timestamp(TimeUnit::Millisecond, None),
            false,
        ),
        Field::new(
            "micro",
            DataType::Timestamp(TimeUnit::Microsecond, None),
            false,
        ),
        Field::new(
            "nano",
            DataType::Timestamp(TimeUnit::Nanosecond, None),
            false,
        ),
    ]);

    let batch = RecordBatch::try_new(
        Arc::new(schema),
        vec![
            Arc::new(sec_array),
            Arc::new(mili_array),
            Arc::new(micro_array),
            Arc::new(nano_array),
        ],
    )
    .unwrap();

    println!("{:?}", batch);
}
```
Output:
```
RecordBatch { schema: Schema { fields: [Field { name: "sec", data_type: Timestamp(Second, None), nullable: false, dict_id: 0, dict_is_ordered: false, metadata: {} }, Field { name: "mili", data_type: Timestamp(Millisecond, None), nullable: false, dict_id: 0, dict_is_ordered: false, metadata: {} }, Field { name: "micro", data_type: Timestamp(Microsecond, None), nullable: false, dict_id: 0, dict_is_ordered: false, metadata: {} }, Field { name: "nano", data_type: Timestamp(Nanosecond, None), nullable: false, dict_id: 0, dict_is_ordered: false, metadata: {} }], metadata: {} }, columns: [PrimitiveArray<Timestamp(Second, None)>
[
  1969-12-31T23:59:59,
], PrimitiveArray<Timestamp(Millisecond, None)>
[
  1969-12-31T23:59:59.999,
], PrimitiveArray<Timestamp(Microsecond, None)>
[
  1969-12-31T23:59:59.999999,
], PrimitiveArray<Timestamp(Nanosecond, None)>
[
  1969-12-31T23:59:59.999999999,
]], row_count: 1 }
```

I don't think that Arrow doesn't support negative timestamp values.

Chrono's DateTime has a from_timestamp_nanos function that can obtain values from nanoseconds. How about modifying **epoch_to_timestamp** as follows?
```
pub fn epoch_to_timestamp<const UNIT_FACTOR: i64>(from: i64) -> Timestamp {
    let timestamp = from * (1_000_000_000 / UNIT_FACTOR);
    let ndt = chrono::DateTime::from_timestamp_nanos(timestamp);
    let date = ndt.date_naive();
    let time = ndt.time();
    Timestamp {
        year: date.year().try_into().unwrap(),
        month: date.month().try_into().unwrap(),
        day: date.day().try_into().unwrap(),
        hour: time.hour().try_into().unwrap(),
        minute: time.minute().try_into().unwrap(),
        second: time.second().try_into().unwrap(),
        fraction: time.nanosecond(),
    }
 }

```

**Comment by pacman82** [comment]
fixed in `arrow-odbc 13.0.1`
